config/v1/types_registries.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Registries holds cluster-wide information about how to handle the registries config.  The canonical name is `cluster`
+type Registries struct {
```

Comment:
Maybe we could just name this `Registry`. I am drawing parallel to the `Build` type. @bparees @derekwaynecarr thoughts?

Reply:
it's a bit awkward either way(it is a single object that represents the cluster's configuration with respect to registries(more than one registry) but it's probably less awkward as a singular so it can be pluralized.

Another option, particularly if this is going to grow to encompass other container runtime behavior, would be to call it ContainerRuntime (containerruntime.config.openshift.io).

Reply:
> Another option, particularly if this is going to grow to encompass other container runtime behavior, would be to call it ContainerRuntime (containerruntime.config.openshift.io).

I could see this being the case, as builds used the preceding config object to generate `registries.conf` and `policy.json` files for the container runtime.

Reply:
Renamed to `Registry`

Reply:
We have a ContainerRuntimeConfig CRD in the MCO. ContainerRuntime sounds good to me as long as it won't be confusing for the users to have a CRD with a similar name in the MCO and global configs.

Reply:
I think there is a non-trivial risk of confusion, since MCO and `config.openshift.io` are both "configuration related" namespaces. Would changes to this object lead to action from the MCO itself?

I'm less of a fan of `Registry` - I think cluster admins would be more likely assume this manages configuration for the integrated image registry.

Reply:
@adambkaplan if you think there's likely to be confusion if we use ContainerRuntime and if you don't like Registry, do you have a third suggestion? :)

@umohnani8 do we need to retain the containerruntimeconfig CRD?  should it be entirely replaced with this new api?

Reply:
> do we need to retain the containerruntimeconfig CRD? should it be entirely replaced with this new api?

I like the idea of making this replace the MCO's CRD. In that case I'd have `containerruntimes.config.openshift.io` as the top level, with `Registries` as a config within `spec`:

```
spec:
  registries:
    allowed: []
    blocked: []
    insecure: []
```

Reply:
(go type should probably be named `RuntimeRegistries or `ContainerRuntimeRegistries`)

Reply:
Yeah we need to retain the containerruntimeconfig CRD in the MCO, since that allows the users to configure certain options in crio.conf and stroage.conf but only for specific machine sets and not the whole cluster.
We need the registries configuration for both the builder pods/container runtime and can be cluster wide, hence the global config approach.
@mrunalp please correct me if I am mistaken.

Reply:
what api group is the MCO CRD in?

Reply:
@bparees it is in `machineconfiguration.openshift.io`

Reply:
then imho it would be ok to have another ContainerRuntime resource in config.openshift.io, but it probably needs @openshift/api-reviewers to weigh in.